src/full/Agda/Compiler/JS/Pretty.hs

```diff
@@ -301,6 +301,21 @@ instance Pretty [(GlobalId, Export)] where
            | (g, Export ls e) <- es ]
 
 instance Pretty Module where
+  pretty opt@(n, min, JSES6) (Module m is es callMain) = vsep
+    [ "import agdaRTS from \"./agda-rts.mjs\";"
+    , imports
+    , "const exports = {};"
+    , exports opt Set.empty es
+    , pretty opt callMain
+    , ";export default exports;"
```

Comment:
This works, but it is a bit unfortunate that this is the approach that had to be taken. The approach here effectively boils down to the same approach that is taken with cjs compilation mode. This negates tree shaking - one of the main features of es6 modules for browser builds, where the rest of the node.js tooling can omit unused functions in the bundled output.

Reply:
Thanks for the review.

The `exports` table keeps Agda names readable, which helps debugging.
e.g. `z_jAgda_Agda_Builtin_Sigma["_,_"]` is more human-friendly than `z_jAgda_Agda_Builtin_Sigma_zuz2cUzu`.

I suggest postponing performance optimization until more data is available -- perhaps performance will be OK even without tree-shaking.